wasi-threads.wit.md

```diff
@@ -0,0 +1,41 @@
+# WASI threads API
+
+WASI threads is an API for thread creation.
+
+It's goal is to provide functions that allow implementation of a subset of `pthreads` API, but it doesn't aim to be 100% compatible with POSIX threads standard.
+
+
+## thread-id
+
+```wit
+/// Unique thread identifier.
+type thread-id = u32
+```
+
+## start-arg
+
+```wit
+/// A reference to data passed to the start function (`wasi_thread_start()`) called by the newly spawned thread.
+resource start-arg {
+}
```

Comment:
Is resource the right thing to use for `start-arg` here?   What we want is really `void*` right?   When we have actual unforgable resource handler will be switching to those as the start arg?  Its not clear to me that we will.

In fact. would it make more sense for `thread-id` to be a resource since its much closer to a file handle which is that canonical use case of `resource` IIUC. 

For this initial commit, perhaps we could just use u32 for both and discuss further in followups?